src/auth/channel.ts

```diff
@@ -0,0 +1,73 @@
+import * as did from "../did/index.js"
+import { setup } from "../setup/internal.js"
+import { LinkingError } from "./linking.js"
+
+import type { Maybe } from "../common/index.js"
+
+export type Channel = {
+  send: (data: ChannelData) => void
+  close: () => void
+}
+
+export type ChannelOptions = {
+  username: string
+  handleMessage: (event: MessageEvent) => void
+}
+
+type ChannelData = string | ArrayBufferLike | Blob | ArrayBufferView
+
+export const createChannel = async (options: ChannelOptions): Promise<Channel> => {
```

Comment:
What do you think of calling this `createWssChannel`? I think it might be more accurate, and I liked the naming of `closeWssChannel` below, although I know it doesn't match the dependency-injected function's naming.

Reply:
Agreed. Renaming this is also great because it makes it more explicit that the default and lobby implementations are using websockets.